### PR TITLE
add tool for supervise python api

### PR DESCRIPTION
**Pull request by guochaorong**
resolve https://github.com/PaddlePaddle/Paddle/pull/11556

### CODE REVIEW

File: tools/check_api.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    fail_list = []
+    for file_name in get_all_files(base_dir):
+        if file_name.split('/')[4] == 'tests':
+            continue
+        if file_name.split('/')[-1].endswith('__.py'):
+            continue
+        module_str = '.'.join(file_name.split('/')[3:])
+        module_str = '.'.join(module_str.split('.')[:-1])
+        print module_str
+        args = parse_args()
```
Comment (panyx0718) [inline tools/check_api.py:160]: perhaps we only need to parse_args once at the beginning?

File: tools/check_api.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        return
+    if not os.path.exists(dest_dir):
+        os.mkdir(dest_dir)
+    with open('%s/%s' % (dest_dir, module_str), 'w') as f:
+        f.write(json.dumps(args_dict))
+
+
+def check_api(module_str, dest_dir):
+    '''get current api's info, 
+        then compare with info loaded from saved'''
```
Comment (panyx0718) [inline tools/check_api.py:126]: you can put ``` at the next line.
Would be good to use the same code style as paddle-core's code.

**Review by panyx0718 — APPROVED** [review]
LG. Do you have an example output?

File: tools/check_api.py (hunk 1 above)
Comment (guochaorong) [reply tools/check_api.py:160]: yes , it will be clearer

File: tools/check_api.py (hunk 2 above)
Comment (guochaorong) [reply tools/check_api.py:126]: got it!

**Comment by guochaorong** [comment]
output are as below:
{
	"ParallelExecutor.__init__": {
		"keywords": "kwargs",
		"args": ["self", "use_cuda", "loss_name", "main_program", "share_vars_from", "exec_strategy", "build_strategy", "num_trainers", "trainer_id"],
		"defaults": "(None, None, None, None, None, 1, 0)",
		"varargs": null
	},
	"ParallelExecutor.bcast_params": {
		"keywords": null,
		"args": ["self"],
		"defaults": "None",
		"varargs": null
	},
	"ParallelExecutor.run": {
		"keywords": null,
		"args": ["self", "fetch_list", "feed", "feed_dict"],
		"defaults": "(None, None)",
		"varargs": null
	}
}

if api changed,  error are as below:
Exception: api changed: ParallelExecutor.__init__ 

before: {
	u 'keywords': u 'kwargs',
	u 'args': [u 'self', u 'use_cu**', u 'loss_name', u 'main_program', u 'share_vars_from', u 'exec_strategy', u 'build_strategy', u 'num_trainers', u 'trainer_id'],
	u 'defaults': u '(None, None, None, None, None, 1, 0)',
	u 'varargs': None
}
now: {
	'keywords': 'kwargs',
	'args': ['self', 'use_cuda', 'loss_name', 'main_program', 'share_vars_from', 'exec_strategy', 'build_strategy', 'num_trainers', 'trainer_id'],
	'defaults': '(None, None, None, None, None, 1, 0)',
	'varargs': None
}

**Comment by guochaorong** [comment]
oh， 想起来，于洋老师改了一版优雅版的交了~
https://github.com/PaddlePaddle/Paddle/pull/11573/files

**Pull request closed without merging**